source/MREAlgorithms/MREPointCloudTriangulationHelpers.cpp

```diff
@@ -35,16 +55,35 @@ namespace MRE
 using namespace MR;
 namespace TriangulationHelpers
 {
+template<typename T>
+std::list<T>::const_iterator cycleNext( const std::list<T>& list, const typename std::list<T>::const_iterator& it )
```

Comment:
std::list is very inefficient in most cases compared to std::vector even if insert elements inside. Have you tested std::vector instead?